pkg/apis/scheduling/v1/helpers.go

```diff
@@ -14,43 +14,62 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package scheduling
+package v1
 
 import (
 	"fmt"
+	"k8s.io/api/scheduling/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/scheduling"
 )
 
 // SystemPriorityClasses define system priority classes that are auto-created at cluster bootstrapping.
 // Our API validation logic ensures that any priority class that has a system prefix or its value
 // is higher than HighestUserDefinablePriority is equal to one of these SystemPriorityClasses.
-var systemPriorityClasses = []*PriorityClass{
+var systemPriorityClasses = []*v1.PriorityClass{
 	{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: SystemNodeCritical,
+			Name: scheduling.SystemNodeCritical,
 		},
-		Value:       SystemCriticalPriority + 1000,
+		Value:       scheduling.SystemCriticalPriority + 1000,
 		Description: "Used for system critical pods that must not be moved from their current node.",
 	},
 	{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: SystemClusterCritical,
+			Name: scheduling.SystemClusterCritical,
 		},
-		Value:       SystemCriticalPriority,
+		Value:       scheduling.SystemCriticalPriority,
 		Description: "Used for system critical pods that must run in the cluster, but can be moved to another node if necessary.",
 	},
 }
 
 // SystemPriorityClasses returns the list of system priority classes.
 // NOTE: be careful not to modify any of elements of the returned array directly.
-func SystemPriorityClasses() []*PriorityClass {
+func SystemPriorityClasses() []*v1.PriorityClass {
 	return systemPriorityClasses
 }
 
 // IsKnownSystemPriorityClass checks that "pc" is equal to one of the system PriorityClasses.
 // It ignores "description", labels, annotations, etc. of the PriorityClass.
-func IsKnownSystemPriorityClass(pc *PriorityClass) (bool, error) {
-	for _, spc := range systemPriorityClasses {
+func IsKnownSystemPriorityClass(pc *v1.PriorityClass) (bool, error) {
+	for _, spc := range SystemPriorityClasses() {
+		if spc.Name == pc.Name {
+			if spc.Value != pc.Value {
+				return false, fmt.Errorf("value of %v PriorityClass must be %v", spc.Name, spc.Value)
+			}
+			if spc.GlobalDefault != pc.GlobalDefault {
+				return false, fmt.Errorf("globalDefault of %v PriorityClass must be %v", spc.Name, spc.GlobalDefault)
+			}
+			return true, nil
+		}
+	}
+	return false, fmt.Errorf("%v is not a known system priority class", pc.Name)
+}
+
+// IsKnownSystemInternalPriorityClass checks that "pc" is equal to one of the system PriorityClasses.
+// It ignores "description", labels, annotations, etc. of the PriorityClass.
+func IsKnownSystemInternalPriorityClass(pc *scheduling.PriorityClass) (bool, error) {
```

Comment:
Why do we need to distinguish "internal" from other system priority classes?

Reply:
it's the validation of priority-class using the internal helpers so we can't completely strip this.

Reply:
I am not sure if I understand what problem you are trying to solve here. It looks like `IsKnownSystemPriorityClass` is no longer used. Couldn't have we made the necessary changes to  `IsKnownSystemPriorityClass` and not added `IsKnownSystemInternalPriorityClass`?

Reply:
the both is being used. i'm trying to avoid doing explicit conversion between internal types and versioned types. 

`IsKnownSystemPriorityClass` is in the post-start-hook, and `IsKnownSystemInternalPriorityClass` is in the validation [https://github.com/kubernetes/kubernetes/blob/eaf89cfbb483f99411141f70c294e8a014e49cec/pkg/apis/scheduling/validation/validation.go#L37](https://github.com/kubernetes/kubernetes/blob/eaf89cfbb483f99411141f70c294e8a014e49cec/pkg/apis/scheduling/validation/validation.go#L37)

Reply:
Alternately, make IsKnownSystemPriorityClass take the name, value, globalDefault and preemptionPolicy fields as parameters, rather than the entire object. Then both callers can collapse onto a single method